daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -227,6 +231,31 @@ runLedgerUploadDar flags darPathM = do
       exitFailure
     Right () -> putStrLn "DAR upload succeeded."
 
+-- | Upload a DAR file to the ledger. Token is given instead of inferred from LedgerFlags. (Defaults to project DAR)
+runLedgerUploadDarWithToken :: String -> LedgerFlags -> Maybe FilePath -> IO ()
```

Comment:
what do you think of slicing this slightly differently such that we can reduce the duplication a bit? I made a draft PR against your branch with my suggestion https://github.com/digital-asset/daml/pull/14191/files

Reply:
the problem is with `getDefaultArgs` and especially with `getProjectLedgerAccessToken`
https://github.com/digital-asset/daml/blob/18ba9db66cc35523d27f09de868a59f752bfecfe/daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs#L146 since the token file is not created yet and cannot be found in the project directory.

Reply:
I’m not quite sure I understand the issue. I would have expected that `getProjectLedgerAccessToken` returns `Nothing` and then we override the token in the args. Do you have a test that would break by my change?

Reply:
changed my PR to target `main` to kickoff CI so we can see what if anything fails.

Reply:
I understood your point now, I’ve updated my PR which should now work. I’ll leave it up to you to decide which solution you prefer. I am happy either way.

Reply:
if you do want to include it just cherry pick the commit to your branch o something.

Reply:
Yes I will cherry pick it when the CI succeeds. Thanks.